cse/src/main/java/com/adyen/checkout/cse/internal/JSONWebEncryptor.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 27/9/2023.
+ */
+
+package com.adyen.checkout.cse.internal
+
+import com.adyen.checkout.cse.EncryptionException
+import org.json.JSONObject
+import java.math.BigInteger
+import java.nio.ByteBuffer
+import java.security.AlgorithmParameters
+import java.security.InvalidKeyException
+import java.security.KeyFactory
+import java.security.NoSuchAlgorithmException
+import java.security.PublicKey
+import java.security.SecureRandom
+import java.security.spec.InvalidKeySpecException
+import java.security.spec.MGF1ParameterSpec
+import java.security.spec.RSAPublicKeySpec
+import javax.crypto.Cipher
+import javax.crypto.IllegalBlockSizeException
+import javax.crypto.Mac
+import javax.crypto.NoSuchPaddingException
+import javax.crypto.SecretKey
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.OAEPParameterSpec
+import javax.crypto.spec.PSource
+import javax.crypto.spec.SecretKeySpec
+
+@Suppress("TooManyFunctions")
+internal class JSONWebEncryptor {
+
+    private val keyFactory: KeyFactory = try {
+        KeyFactory.getInstance(RSA_ALGORITHM)
+    } catch (e: NoSuchAlgorithmException) {
+        throw EncryptionException("RSA KeyFactory not found.", e)
+    }
+
+    fun encrypt(publicKey: String, payload: String): String {
+        val pubKey = generatePublicKey(publicKey)
+        val cek = generateCEK()
+        val encryptedKey = Base64String(encryptCEK(pubKey, cek))
+        val jweObject = encrypt(payload, cek, encryptedKey)
+        return serialize(jweObject)
+    }
+
+    private fun generatePublicKey(publicKey: String): PublicKey {
+        val keyComponents = publicKey.split("|")
```

Comment:
We were using `ValidationUtils.isPublicKeyValid` in the previous implementation, could we use it again?